File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBInteger.cs
using System;
using System.Numerics;

namespace se.sics.prologbeans
{
	
	/// <summary> <code>PBInteger</code> is the representation of Prolog integers.</summary>
	
	class PBInteger:PBAtomic
	{
		override internal int Type {
			get {
				return INTEGER;
			}
			
		}

		override public bool Integer {
			get {
				return true;
			}
		}

		override sealed public bool Number {
			get {
				return true;
			}
		}

		internal long longValue;

		/// <summary> Creates a new <code>PBInteger</code> instance with the specified value.</summary>
		internal PBInteger (long value) : base (Convert.ToString (value))
		{
			longValue = value;
		}

		internal PBInteger (long value, string name) : base (name)
		{
			longValue = value;
		}

		public override long intValue ()
		{
			return longValue;
		}

		public override BigInteger bigIntegerValue ()
		{
			return new BigInteger(intValue());
		}

		internal override string toPrologString ()
		{
			return ToString ();
		}

		public override string ToString ()
		{
			return Convert.ToString (longValue);
		}
	}
}

File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBInteger.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

/**
 * <code>PBInteger</code> is the representation of Prolog integers.
 */

class PBInteger extends PBAtomic
{
    long longValue;

    /**
     * Creates a new <code>PBInteger</code> instance with the specified value.
     */
    PBInteger(long value)
    {
        super(Long.toString(value));
        longValue = value;
    }

    PBInteger(long value, String name)
    {
        super(name);
        longValue = value;
    }

    int getType()
    {
        return INTEGER;
    }

    public boolean isInteger()
    {
        return true;
    }

    public long intValue()
    {
        return longValue;
    }

    String toPrologString()
    {
        return toString();
    }

    public String toString()
    {
        return Long.toString(longValue);
    }

}
